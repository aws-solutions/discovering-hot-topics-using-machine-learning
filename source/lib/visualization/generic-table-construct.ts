#!/usr/bin/env node
/**********************************************************************************************************************
* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
* SPDX-License-Identifier: Apache-2.0
 *********************************************************************************************************************/

import * as glue from 'aws-cdk-lib/aws-glue';
import * as glue_alpha from '@aws-cdk/aws-glue-alpha';
import * as s3 from 'aws-cdk-lib/aws-s3';
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

export interface GenericCfnTableProps {
    readonly tableName: string;
    readonly database: glue_alpha.IDatabase,
    readonly s3InputDataBucket: s3.Bucket,
    readonly s3BucketPrefix: string,
}

export abstract class GenericCfnTable extends Construct {
    public readonly table: glue.CfnTable;

    constructor(scope: Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id);

        this.table = new glue.CfnTable(this, `Cfn${props.tableName}`, {
            catalogId: cdk.Aws.ACCOUNT_ID,
            databaseName: props.database.databaseName,
            tableInput: {
                description: `A table created with partition projection for ${props.tableName}`,
                name: props.tableName,
                parameters: {
                    "classification": "parquet",
                    "has_encryped_data": false,
                    "projection.enabled": "TRUE",
                    "projection.created_at.range": "NOW-45DAYS,NOW",
                    "projection.created_at.type": "date",
                    "projection.created_at.format": "yyyy-MM-dd"
                },
                partitionKeys: this.partitionKeys,
                storageDescriptor: {
                    columns: this.getColumns(),
                    compressed: false,
                    inputFormat: "org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat",
                    location: `${props.s3InputDataBucket.s3UrlForObject(props.s3BucketPrefix)}`,
                    outputFormat: "org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat",
                    serdeInfo: {
                        serializationLibrary: "org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe"
                    },
                    storedAsSubDirectories: true
                },
                tableType: 'EXTERNAL_TABLE',
            }
        });
    }

    protected abstract getColumns(): glue.CfnTable.ColumnProperty[];

    protected get partitionKeys(): glue.CfnTable.ColumnProperty[] {
        return [{
            name: 'created_at',
            type: glue_alpha.Schema.TIMESTAMP.inputString
        }]
    }

    protected get coreColumns() {
        return [{
            name: 'account_name',
            type: glue_alpha.Schema.STRING.inputString
        }, {
            name: 'platform',
            type: glue_alpha.Schema.STRING.inputString
        }, {
            name: 'search_query',
            type: glue_alpha.Schema.STRING.inputString
        }, {
            name: 'id_str',
            type: glue_alpha.Schema.STRING.inputString
        }]
    }
}