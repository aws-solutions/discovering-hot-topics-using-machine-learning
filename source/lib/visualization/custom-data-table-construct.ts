/**********************************************************************************************************************
 *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.                                           *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import * as glue from '@aws-cdk/aws-glue';
import * as cdk from '@aws-cdk/core';
import { GenericCfnTable, GenericCfnTableProps } from './generic-table-construct';

/**
 * Normalizing the data structure of the JSON object to a relational form and remove arrays since arrays
 * as a datatype is not supported by Amazon QuickSight.
 */
export class CustomDataTable extends GenericCfnTable {
    constructor(scope: cdk.Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id, props);
    }

    protected getColumns(): glue.CfnTable.ColumnProperty[] {
        return [
            {
                name: 'account_name',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'platform',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'id_str',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'parent_id',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'text',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'lang',
                type: glue.Schema.STRING.inputString
            },
            {
                name: '_translated_text',
                type: glue.Schema.STRING.inputString
            },
            {
                name: '_cleansed_text',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'source_file',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'Id',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'BeginOffsetMillis',
                type: glue.Schema.BIG_INT.inputString
            },
            {
                name: 'EndOffsetMillis',
                type: glue.Schema.BIG_INT.inputString
            },
            {
                name: 'Sentiment',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'ParticipantRole',
                type: glue.Schema.STRING.inputString
            }
        ];
    }
}

export class LoudnessScoreTable extends GenericCfnTable {
    constructor(scope: cdk.Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id, props);
    }

    protected getColumns(): glue.CfnTable.ColumnProperty[] {
        return [
            {
                name: 'score',
                type: glue.Schema.FLOAT.inputString
            },
            {
                name: 'account_name',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'platform',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'id_str',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'parent_id',
                type: glue.Schema.STRING.inputString
            }
        ];
    }
}

export class ItemTable extends GenericCfnTable {
    constructor(scope: cdk.Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id, props);
    }

    protected getColumns(): glue.CfnTable.ColumnProperty[] {
        return [
            {
                name: 'account_name',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'platform',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'id_str',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'parent_id',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'Type',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'Confidence',
                type: glue.Schema.FLOAT.inputString
            },
            {
                name: 'Content',
                type: glue.Schema.STRING.inputString
            },
            {
                name: 'BeginOffsetMillis',
                type: glue.Schema.BIG_INT.inputString
            },
            {
                name: 'EndOffsetMillis',
                type: glue.Schema.BIG_INT.inputString
            }
        ];
    }
}
