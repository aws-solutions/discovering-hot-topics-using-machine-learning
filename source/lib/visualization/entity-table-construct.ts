#!/usr/bin/env node
/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import * as glue from 'aws-cdk-lib/aws-glue';
import * as glue_alpha  from '@aws-cdk/aws-glue-alpha';
import { Construct } from 'constructs';
import { GenericCfnTable, GenericCfnTableProps } from "./generic-table-construct";

export class EntityTable extends GenericCfnTable {
    constructor(scope: Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id, props);
    }

    protected getColumns(): glue.CfnTable.ColumnProperty[] {
        return [
            ...this.coreColumns, {
                name: 'text',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'translated_text',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'entity_text',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'entity_type',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'entity_score',
                type: glue_alpha.Schema.DOUBLE.inputString
            }, {
                name: 'entity_begin_offset',
                type: glue_alpha.Schema.INTEGER.inputString
            }, {
                name: 'entity_end_offset',
                type: glue_alpha.Schema.INTEGER.inputString
            }];
    }
}
