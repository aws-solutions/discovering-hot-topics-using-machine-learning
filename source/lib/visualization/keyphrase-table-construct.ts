#!/usr/bin/env node
/**********************************************************************************************************************
* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
* SPDX-License-Identifier: Apache-2.0
 *********************************************************************************************************************/

import * as glue_alpha from '@aws-cdk/aws-glue-alpha';
import * as glue from 'aws-cdk-lib/aws-glue';
import { Construct } from 'constructs';
import { GenericCfnTable, GenericCfnTableProps } from "./generic-table-construct";


export class KeyPhraseTable extends GenericCfnTable {

    constructor(scope: Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id, props);
    }

    protected getColumns(): glue.CfnTable.ColumnProperty[] {

        return [
            ...this.coreColumns, {
                name: 'text',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'translated_text',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'phrase',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'phrase_score',
                type: glue_alpha.Schema.DOUBLE.inputString
            }, {
                name: 'phrase_begin_offset',
                type: glue_alpha.Schema.INTEGER.inputString
            }, {
                name: 'phrase_end_offset',
                type: glue_alpha.Schema.INTEGER.inputString
            }];
    }
}
