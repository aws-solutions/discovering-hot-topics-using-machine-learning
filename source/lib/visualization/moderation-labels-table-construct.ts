#!/usr/bin/env node
/**********************************************************************************************************************
* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
* SPDX-License-Identifier: Apache-2.0
 *********************************************************************************************************************/

import * as glue_alpha from '@aws-cdk/aws-glue-alpha';
import * as glue from 'aws-cdk-lib/aws-glue';
import { Construct } from 'constructs';
import { GenericCfnTable, GenericCfnTableProps } from "./generic-table-construct";

export class ModerationLabelsTable extends GenericCfnTable {

    constructor(scope: Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id, props);
    }

    protected getColumns(): glue.CfnTable.ColumnProperty[] {
        return [
            ...this.coreColumns, {
                name: 'image_url',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'label_name',
                type: glue_alpha.Schema.STRING.inputString
            }, {
                name: 'confidence',
                type: glue_alpha.Schema.DOUBLE.inputString
            }];
    }
}
