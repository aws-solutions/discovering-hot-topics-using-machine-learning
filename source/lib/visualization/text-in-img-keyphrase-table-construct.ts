#!/usr/bin/env node
/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import * as glue from '@aws-cdk/aws-glue';
import * as cdk from '@aws-cdk/core';
import { GenericCfnTable, GenericCfnTableProps } from './generic-table-construct';
export class TextInImgKeyPhraseTable extends GenericCfnTable {

    constructor(scope: cdk.Construct, id: string, props: GenericCfnTableProps) {
        super(scope, id, props);
    }

    protected getColumns(): glue.CfnTable.ColumnProperty[] {
        return [
            ...this.coreColumns, {
                name: 'text',
                type: glue.Schema.STRING.inputString
            }, {
                name: 'phrase',
                type: glue.Schema.STRING.inputString
            }, {
                name: 'phrase_score',
                type: glue.Schema.DOUBLE.inputString
            }, {
                name: 'phrase_begin_offset',
                type: glue.Schema.INTEGER.inputString
            }, {
                name: 'phrase_end_offset',
                type: glue.Schema.INTEGER.inputString
            }, {
                name: 'image_url',
                type: glue.Schema.STRING.inputString
            }];
    }
}
