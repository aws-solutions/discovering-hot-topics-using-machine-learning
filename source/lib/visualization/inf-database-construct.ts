#!/usr/bin/env node
/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import * as glue from 'aws-cdk-lib/aws-glue';
import * as glue_alpha from '@aws-cdk/aws-glue-alpha';
import * as s3 from 'aws-cdk-lib/aws-s3';
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { CustomDataTable, LoudnessScoreTable, ItemTable } from './custom-data-table-construct';
import { EntityTable } from './entity-table-construct';
import { KeyPhraseTable } from './keyphrase-table-construct';
import { ModerationLabelsTable } from './moderation-labels-table-construct';
import { NewsFeedTable } from './newsfeed-table-construct';
import { SentimentTable } from './sentiment-table-construct';
import { TextInImgEntityTable } from './text-in-image-entities-table-construct';
import { TextInImgKeyPhraseTable } from './text-in-img-keyphrase-table-construct';
import { TextInImgSentimentTable } from './text-in-img-sentiment-table-construct';
import { TopicMappingsTable } from './topicmappings-table-construct';
import { TopicsTable } from './topics-table-construct';
import { TwitterTable } from './twitter-table-construct';
import { YoutubeCommentsTable } from './youtubecomments-table-construct';
import { MetdataTable } from './metadata-table-construct';
import { RedditCommentsTable } from './reddit-comments-table-construct';

export interface InferenceDatabaseProps {
    readonly s3InputDataBucket: s3.Bucket;
    readonly tablePrefixMappings: Map<string, string>;
    readonly s3LoggingBucket: s3.Bucket;
}
export class InferenceDatabase extends Construct {
    public readonly database: glue_alpha.Database;
    public readonly tableMap: Map<any, glue.CfnTable>;
    public readonly glueKMSKey: string;

    constructor(scope: Construct, id: string, props: InferenceDatabaseProps) {
        super(scope, id);

        this.glueKMSKey = `arn:${cdk.Aws.PARTITION}:kms:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:alias/aws/glue`;

        new glue.CfnSecurityConfiguration(this, 'GlueSecConfig', {
            name: 'socialmediadb-sec-config',
            encryptionConfiguration: {
                cloudWatchEncryption: {
                    cloudWatchEncryptionMode: 'SSE-KMS',
                    kmsKeyArn: this.glueKMSKey
                },
                s3Encryptions: [
                    {
                        s3EncryptionMode: 'SSE-S3'
                    }
                ],
                jobBookmarksEncryption: {
                    jobBookmarksEncryptionMode: 'CSE-KMS',
                    kmsKeyArn: this.glueKMSKey
                }
            }
        });

        this.database = new glue_alpha.Database(this, 'TweetDB', {
            databaseName: 'socialmediadb'
        });

        new glue.CfnDataCatalogEncryptionSettings(this, 'TweetDBEncryption', {
            catalogId: this.database.catalogId,
            dataCatalogEncryptionSettings: {
                encryptionAtRest: {
                    catalogEncryptionMode: 'SSE-KMS',
                    sseAwsKmsKeyId: 'alias/aws/glue'
                }
            }
        });

        this.tableMap = new Map();
        const sentimentTable = new SentimentTable(this, 'Sentiment', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('Sentiment')!}/`,
            tableName: props.tablePrefixMappings.get('Sentiment')!,
            database: this.database
        });
        this.tableMap.set('Sentiment', sentimentTable.table);

        const entityTable = new EntityTable(this, 'Entity', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('Entity')!}/`,
            tableName: props.tablePrefixMappings.get('Entity')!,
            database: this.database
        });
        this.tableMap.set('Entity', entityTable.table);

        const keyphraseTable = new KeyPhraseTable(this, 'KeyPhrase', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('KeyPhrase')!}/`,
            tableName: props.tablePrefixMappings.get('KeyPhrase')!,
            database: this.database
        });
        this.tableMap.set('KeyPhrase', keyphraseTable.table);

        const topicsTable = new TopicsTable(this, 'Topics', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('Topics')!}/`,
            tableName: props.tablePrefixMappings.get('Topics')!,
            database: this.database
        });
        this.tableMap.set('Topics', topicsTable.table);

        const topicMappingsTable = new TopicMappingsTable(this, 'TopicMappings', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('TopicMappings')!}/`,
            tableName: props.tablePrefixMappings.get('TopicMappings')!,
            database: this.database
        });
        this.tableMap.set('TopicMappings', topicMappingsTable.table);

        const txtInImgEntityTable = new TextInImgEntityTable(this, 'TxtInImgEntity', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('TxtInImgEntity')!}/`,
            tableName: props.tablePrefixMappings.get('TxtInImgEntity')!,
            database: this.database
        });
        this.tableMap.set('TxtInImgEntity', txtInImgEntityTable.table);

        const txtInImgKeyPhraseTable = new TextInImgKeyPhraseTable(this, 'TxtInImgKeyPhrase', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('TxtInImgKeyPhrase')!}/`,
            tableName: props.tablePrefixMappings.get('TxtInImgKeyPhrase')!,
            database: this.database
        });
        this.tableMap.set('TxtInImgKeyPhrase', txtInImgKeyPhraseTable.table);

        const txtInImgSentiment = new TextInImgSentimentTable(this, 'TxtInImgSentiment', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('TxtInImgSentiment')!}/`,
            tableName: props.tablePrefixMappings.get('TxtInImgSentiment')!,
            database: this.database
        });
        this.tableMap.set('TxtInImgSentiment', txtInImgSentiment.table);

        const moderationLabelsTable = new ModerationLabelsTable(this, 'ModerationLabels', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('ModerationLabels')!}/`,
            tableName: props.tablePrefixMappings.get('ModerationLabels')!,
            database: this.database
        });
        this.tableMap.set('ModerationLabels', moderationLabelsTable.table);

        const twitterTable = new TwitterTable(this, 'Twitter', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('TwFeedStorage')!}/`,
            tableName: props.tablePrefixMappings.get('TwFeedStorage')!,
            database: this.database
        });
        this.tableMap.set('TwFeedStorage', twitterTable.table);

        const newsFeedTable = new NewsFeedTable(this, 'NewsFeed', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('NewsFeedStorage')!}/`,
            tableName: props.tablePrefixMappings.get('NewsFeedStorage')!,
            database: this.database
        });
        this.tableMap.set('NewsFeedStorage', newsFeedTable.table);

        const youtubeCommentsTable = new YoutubeCommentsTable(this, 'YoutubeComments', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('YouTubeComments')!}/`,
            tableName: props.tablePrefixMappings.get('YouTubeComments')!,
            database: this.database
        });
        this.tableMap.set('YouTubeComments', youtubeCommentsTable.table);

        const customIngestionTable = new CustomDataTable(this, 'CustomIngestion', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('CustomIngestion')!}/`,
            tableName: props.tablePrefixMappings.get('CustomIngestion')!,
            database: this.database
        });
        this.tableMap.set('CustomIngestion', customIngestionTable.table);

        const customIngestionLoudnessTable = new LoudnessScoreTable(this, 'CustomIngestionLoudness', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('CustomIngestionLoudness')!}/`,
            tableName: props.tablePrefixMappings.get('CustomIngestionLoudness')!,
            database: this.database
        });
        this.tableMap.set('CustomIngestionLoudness', customIngestionLoudnessTable.table);

        const customIngestionItemTable = new ItemTable(this, 'CustomIngestionItem', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('CustomIngestionItem')!}/`,
            tableName: props.tablePrefixMappings.get('CustomIngestionItem')!,
            database: this.database
        });
        this.tableMap.set('CustomIngestionItem', customIngestionItemTable.table);

        const metadataTable = new MetdataTable(this, 'Metadata', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('Metadata')!}/`,
            tableName: props.tablePrefixMappings.get('Metadata')!,
            database: this.database
        });
        this.tableMap.set('Metadata', metadataTable.table);

        const redditCommentsTable = new RedditCommentsTable(this, 'RedditComments', {
            s3InputDataBucket: props.s3InputDataBucket,
            s3BucketPrefix: `${props.tablePrefixMappings.get('RedditComments')!}/`,
            tableName: props.tablePrefixMappings.get('RedditComments')!,
            database: this.database
        });
        this.tableMap.set('RedditComments', redditCommentsTable.table);
    }
}
