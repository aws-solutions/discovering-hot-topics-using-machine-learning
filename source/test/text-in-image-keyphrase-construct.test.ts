/**********************************************************************************************************************
* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
* SPDX-License-Identifier: Apache-2.0
 *********************************************************************************************************************/


import { Template } from 'aws-cdk-lib/assertions';
import { Database } from '@aws-cdk/aws-glue-alpha';
import { Bucket } from 'aws-cdk-lib/aws-s3';
import * as cdk from 'aws-cdk-lib';
import { TextInImgKeyPhraseTable } from '../lib/visualization/text-in-img-keyphrase-table-construct';


test('test workflow stack', () => {
    const stack = new cdk.Stack();
    new TextInImgKeyPhraseTable (stack, 'TxtImgKeyPhraseTable', {
        s3InputDataBucket: new Bucket(stack, 'TestBucket'),
        s3BucketPrefix: 'imgtxtkeyphrase/',
        database: new Database(stack, 'TestDB', {
            databaseName: 'socialmediadb'
        }),
        tableName: 'imgtxtkeyphrase'
    });
});