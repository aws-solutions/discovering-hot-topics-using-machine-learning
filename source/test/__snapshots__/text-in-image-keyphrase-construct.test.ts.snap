// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test workflow stack 1`] = `
{
  "Resources": {
    "TestBucket560B80BC": {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "TestDB27CDA92F": {
      "Properties": {
        "CatalogId": {
          "Ref": "AWS::AccountId",
        },
        "DatabaseInput": {
          "Name": "socialmediadb",
        },
      },
      "Type": "AWS::Glue::Database",
    },
    "TxtImgKeyPhraseTableCfnimgtxtkeyphraseA3CF0E5D": {
      "Properties": {
        "CatalogId": {
          "Ref": "AWS::AccountId",
        },
        "DatabaseName": {
          "Ref": "TestDB27CDA92F",
        },
        "TableInput": {
          "Description": "A table created with partition projection for imgtxtkeyphrase",
          "Name": "imgtxtkeyphrase",
          "Parameters": {
            "classification": "parquet",
            "has_encryped_data": false,
            "projection.created_at.format": "yyyy-MM-dd",
            "projection.created_at.range": "NOW-45DAYS,NOW",
            "projection.created_at.type": "date",
            "projection.enabled": "TRUE",
          },
          "PartitionKeys": [
            {
              "Name": "created_at",
              "Type": "timestamp",
            },
          ],
          "StorageDescriptor": {
            "Columns": [
              {
                "Name": "account_name",
                "Type": "string",
              },
              {
                "Name": "platform",
                "Type": "string",
              },
              {
                "Name": "search_query",
                "Type": "string",
              },
              {
                "Name": "id_str",
                "Type": "string",
              },
              {
                "Name": "text",
                "Type": "string",
              },
              {
                "Name": "phrase",
                "Type": "string",
              },
              {
                "Name": "phrase_score",
                "Type": "double",
              },
              {
                "Name": "phrase_begin_offset",
                "Type": "int",
              },
              {
                "Name": "phrase_end_offset",
                "Type": "int",
              },
              {
                "Name": "image_url",
                "Type": "string",
              },
            ],
            "Compressed": false,
            "InputFormat": "org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat",
            "Location": {
              "Fn::Join": [
                "",
                [
                  "s3://",
                  {
                    "Ref": "TestBucket560B80BC",
                  },
                  "/imgtxtkeyphrase/",
                ],
              ],
            },
            "OutputFormat": "org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat",
            "SerdeInfo": {
              "SerializationLibrary": "org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe",
            },
            "StoredAsSubDirectories": true,
          },
          "TableType": "EXTERNAL_TABLE",
        },
      },
      "Type": "AWS::Glue::Table",
    },
  },
}
`;
